app/src/main/java/edu/uwstout/p2pchat/FileTransfer/SendDataService.java

```diff
@@ -259,6 +248,48 @@ private void transmitINetAddress(@NonNull final Intent intent)
         }
```

Comment:
Is this missing the `finally` block where the socket connection is closed?
```suggestion
        }
        finally
        {
            if(socket.isConnected())
            {
                try
                {
                    socket.close();
                }
                catch(IOException e)
                {
                    e.printStackTrace();
                }
            }
        }
```

Reply:
Also, since the code for closing the socket is identical between the two transmission methods perhaps this could get moved out into its own method. But that's up to you.